org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java

```diff
@@ -570,6 +638,22 @@ protected JvmTypeReference resolveSynonymType(JvmSynonymTypeReference reference,
 		LightweightTypeReference superType = lightweight.getOwner().getServices().getTypeConformanceComputer().getCommonSuperType(lightweight.getMultiTypeComponents(), lightweight.getOwner());
 		return superType.toJavaCompliantTypeReference();
 	}
+	
+	protected void appendFinallyWithResources(List<XVariableDeclaration> resources, ITreeAppendable b) {
+		for (int i = resources.size() - 1; i >= 0; i--) {
+			XVariableDeclaration res = resources.get(i);
+			String resName = getVarName(res, b);
+			b.newLine().append("if (" + resName + " != null) ").append(resName + ".close();");
```

Comment:
Close may declared exceptions. I assume we need to surround these invocations by a try catch.

Reply:
Follow up issue #415